packages/wrangler/src/module-collection.ts

```diff
@@ -123,9 +123,9 @@ export default function createModuleCollector(props: {
 						{
 							filter: new RegExp(
 								"^(" +
-									[...props.wrangler1xlegacyModuleReferences.fileNames].join(
-										"|"
-									) +
+									[...props.wrangler1xlegacyModuleReferences.fileNames]
+										.map((name) => name.replace(/[*+?^${}()|[\]\\]/g, "\\$&"))
```

Comment:
Do you suppose we should add the `.` back in this regex?

Reply:
That's a good idea, have done so